addons/mail/models/mail_message.py

```diff
@@ -96,12 +96,12 @@ def default_get(self, fields):
         help='Attachments are linked to a document through model / res_id and to the message '
              'through this field.')
     parent_id = fields.Many2one(
-        'mail.message', 'Parent Message', index=True, ondelete='set null',
+        'mail.message', 'Parent Message', index="not null", ondelete='set null',
         help="Initial thread message.")
     child_ids = fields.One2many('mail.message', 'parent_id', 'Child Messages')
     # related document
-    model = fields.Char('Related Document Model', index=True)
-    res_id = fields.Many2oneReference('Related Document ID', index=True, model_field='model')
+    model = fields.Char('Related Document Model', index='not null')
+    res_id = fields.Many2oneReference('Related Document ID', index='not null', model_field='model')
```

Comment:
As we already have a combined index (SELECT indexname FROM pg_indexes WHERE indexname = 'mail_message_model_res_id_idx'), do we need to keep specific indexes at all ?

Reply:
Probably not, I'll change that.